cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/system/StreamSystemMetadataWriter.java

```diff
@@ -31,16 +31,32 @@
 public class StreamSystemMetadataWriter extends AbstractSystemMetadataWriter {
 
   private final StreamConfig config;
+  private final long creationTime;
+  private final String description;
 
-  public StreamSystemMetadataWriter(MetadataStore metadataStore, Id.Stream streamId, StreamConfig config) {
+  public StreamSystemMetadataWriter(MetadataStore metadataStore, Id.Stream streamId, StreamConfig config,
+                                    @Nullable String description) {
+    this(metadataStore, streamId, config, -1, description);
+  }
+
+  public StreamSystemMetadataWriter(MetadataStore metadataStore, Id.Stream streamId, StreamConfig config,
```

Comment:
When stream properties are updated, the other constructor that does not take in create time should be used -  
`public StreamSystemMetadataWriter(MetadataStore metadataStore, Id.Stream streamId, StreamConfig config, @Nullable String description)`

However, I see that we don't update system metadata when stream properties are updated. I'll add that code.